components/places/src/bookmark_sync/record.rs

```diff
@@ -450,4 +497,74 @@ mod tests {
             })
         );
     }
+
+    // XXX - all below 'dateadded' tests only check the 'BookmarkItemRecord' variant.
+    // Is there a sane way to check all? Should we maybe have a specific type
+    // avoiding the need to remember `date_added: Option<i64>` needs the serde
+    // `deserialize_with` attribute?
```

Comment:
That's a good idea...since you're almost all the way there, we could make the `StringOrInt` type public, add a serializer (that always serializes as an integer), and maybe a `From<i64>` implementation.

Reply:
FTR, based on our slack discussion, we decided to not do this.